### PR TITLE
added projectDepartment table

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

File: src/ProjectService.Data/Interfaces/IProjectImageRepository.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 namespace LT.DigitalOffice.ProjectService.Data.Interfaces
 {
     [AutoInject]
-    public interface IImageRepository
+    public interface IProjectImageRepository
```
Comment (VLiamin) [inline src/ProjectService.Data/Interfaces/IProjectImageRepository.cs:10]: Why do you changed this name? In project exist only project's images.

File: src/ProjectService.Data/ProjectDepartmentRepository.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using LT.DigitalOffice.ProjectService.Data.Provider;
+using LT.DigitalOffice.ProjectService.Models.Db;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+
+namespace LT.DigitalOffice.ProjectService.Data
+{
+  public class ProjectDepartmentRepository : IProjectDepartmentRepository
+  {
+
```
Comment (VLiamin) [inline src/ProjectService.Data/ProjectDepartmentRepository.cs:14]: empty line

**Review by VLiamin — APPROVED** [review]

File: src/ProjectService.Business/Commands/Project/Interfaces/IGetProjectCommand.cs [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 using LT.DigitalOffice.Kernel.Attributes;
 using LT.DigitalOffice.Kernel.Responses;
-using LT.DigitalOffice.ProjectService.Models.Dto.Requests.Filters;
+using LT.DigitalOffice.ProjectService.Models.Dto.Requests.Project;
 using LT.DigitalOffice.ProjectService.Models.Dto.Responses;
 
 namespace LT.DigitalOffice.ProjectService.Business.Commands.Project.Interfaces
 {
-    [AutoInject]
+  [AutoInject]
```
Comment (VLiamin) [inline src/ProjectService.Business/Commands/Project/Interfaces/IGetProjectCommand.cs:9]: spaces

File: src/ProjectService.Data/Interfaces/IProjectImageRepository.cs (hunk 1 above)
Comment (MariyaSin) [reply src/ProjectService.Data/Interfaces/IProjectImageRepository.cs:10]: the name of the repo must match the name of the table

File: src/ProjectService.Data.Provider.MsSql.Ef/ProjectServiceDbContext.cs [hunk 4]
```diff
@@ -1,52 +1,53 @@
...
+    public DbSet<DbProject> Projects { get; set; }
+    public DbSet<DbProjectDepartment> ProjectsDepartments { get; set; }
+    public DbSet<DbProjectFile> ProjectsFiles { get; set; }
+    public DbSet<DbProjectUser> ProjectsUsers { get; set; }
+    public DbSet<DbProjectImage> Images { get; set; }
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+      modelBuilder.ApplyConfigurationsFromAssembly(
+          Assembly.Load("LT.DigitalOffice.ProjectService.Models.Db"));
```
Comment (VLiamin) [inline src/ProjectService.Data.Provider.MsSql.Ef/ProjectServiceDbContext.cs:22]: spaces

File: src/ProjectService.Data/ProjectDepartmentRepository.cs (hunk 2 above)
Comment (MariyaSin) [reply src/ProjectService.Data/ProjectDepartmentRepository.cs]: done

File: src/ProjectService.Broker/Requests/DepartmentService.cs [hunk 5]
```diff
@@ -33,18 +33,18 @@ public class DepartmentService : IDepartmentService
...
-      List<Guid> projectsIds = null,
+      List<Guid> departmentsIds = null,
       List<Guid> usersIds = null)
     {
       string key;
 
       if (usersIds != null && usersIds.Any())
       {
-        key = usersIds.GetRedisCacheHashCode(projectsIds.FirstOrDefault());
+        key = usersIds.GetRedisCacheHashCode(departmentsIds.FirstOrDefault());
```
[the remark this answers is not in the document]
Comment (ndoriomedov) [reply src/ProjectService.Broker/Requests/DepartmentService.cs:43]: in department service's consumer key is generated from usersIds and departmentsIds
image

File: src/ProjectService.Business/Commands/Department/EditProjectDepartmentCommand.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+      }
+
+      if (!await _departmentRepository.EditAsync(request.ProjectId, request.DepartmentId))
+      {
+        await _departmentRepository.CreateAsync(
+          _mapper.Map(request.ProjectId, request.DepartmentId.Value));
+      }
+
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
```
Comment (ndoriomedov) [inline src/ProjectService.Business/Commands/Department/EditProjectDepartmentCommand.cs:51]: why 201 in edit command?

File: src/ProjectService.Data/ProjectDepartmentRepository.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        return false;
+      }
+
+      dbProjectDepartment.DepartmentId = departmentId.HasValue ? departmentId.Value : dbProjectDepartment.DepartmentId;
+      dbProjectDepartment.IsActive = departmentId.HasValue ? true : false;
+      dbProjectDepartment.CreatedBy = _httpContextAccessor.HttpContext.GetUserId();
+
+      await _provider.SaveAsync();
+
+      return true;
```
Comment (ndoriomedov) [inline src/ProjectService.Data/ProjectDepartmentRepository.cs:54]: return Task

File: src/ProjectService.Data/ProjectDepartmentRepository.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      if (request is null)
+      {
+        return null;
+      }
+
+      _provider.ProjectsDepartments.Add(request);
+      await _provider.SaveAsync();
+
+      return request.Id;
+    }
```
Comment (ndoriomedov) [inline src/ProjectService.Data/ProjectDepartmentRepository.cs:36]: return Task

**Review by ndoriomedov — APPROVED** [review]

File: src/ProjectService.Data.Provider.MsSql.Ef/ProjectServiceDbContext.cs (hunk 4 above)
Comment (MariyaSin) [reply src/ProjectService.Data.Provider.MsSql.Ef/ProjectServiceDbContext.cs:22]: done

File: src/ProjectService.Broker/Requests/DepartmentService.cs [hunk 9]
```diff
@@ -33,18 +33,18 @@ public class DepartmentService : IDepartmentService
 
     public async Task<List<DepartmentData>> GetDepartmentsAsync(
       List<string> errors,
-      List<Guid> projectsIds = null,
+      List<Guid> departmentsIds = null,
       List<Guid> usersIds = null)
     {
       string key;
 
       if (usersIds != null && usersIds.Any())
```
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/ProjectService.Broker/Requests/DepartmentService.cs:41]: done

File: src/ProjectService.Broker/Requests/DepartmentService.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/ProjectService.Broker/Requests/DepartmentService.cs:43]: it's out of the question now)

File: src/ProjectService.Broker/Requests/DepartmentService.cs [hunk 10]
```diff
@@ -33,18 +33,18 @@ public class DepartmentService : IDepartmentService
...
-        key = usersIds.GetRedisCacheHashCode(projectsIds.FirstOrDefault());
+        key = usersIds.GetRedisCacheHashCode(departmentsIds.FirstOrDefault());
       }
-      else if (projectsIds != null && projectsIds.Any())
+      else if (departmentsIds != null && departmentsIds.Any())
       {
-        key = projectsIds.GetRedisCacheHashCode();
+        key = departmentsIds.GetRedisCacheHashCode();
       }
       else
```
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/ProjectService.Broker/Requests/DepartmentService.cs:49]: i do not think so))

**Pull request merged**